cmd/maya-apiserver/spc-watcher/lease.go

```diff
@@ -13,91 +13,48 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+
 package spc
 
 import (
-	"fmt"
-	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
 	clientset "github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
-	menv "github.com/openebs/maya/pkg/env/v1alpha1"
-	"k8s.io/apimachinery/pkg/util/runtime"
-	"strings"
+	"k8s.io/client-go/kubernetes"
 )
 
-const (
-	// SpcLeaseKey is the key that will be used to acquire lease on spc object.
-	// It will be present in spc annotations
-	// If key has an empty value, that means no one has acquired a lease on spc object.
-	SpcLeaseKey = "openebs.io/spc-create-lease"
-)
+// ToDo : Move this file to pkg
+// This struct will be used as a value of lease key that will
+// give information about an acquired lease on object
+// The struct object will be parsed to string which will be then
+// put as a value to the lease key of object annotation.
+type lease struct {
+	// Holder is the namespace/name of the pod who acquires the lease
+	Holder           string `json:"holder"`
+	LeaderTransition int    `json:"leaderTransition"`
```

Comment:
what does leaderTransistion contain? Is it a state in which the Holder is?

Reply:
Also, can you combine these values into the below - Lease struct.

Reply:
`Lease` struct is used to form a object that is in turn used to call the methods of lease interface.
Whereas `lease` is the struct that is parsed into json to be put as value for object lease annotation.
Reason behind not combining  them is the `lease` struct cannot be filled by caller code , it depends which pod gets a chance to acquire a lease and then podName,etc for that pod is filled in that struct.
But naming here is `Lease` and `lease` and is confusing.... Thinking on the names.